test/exchangeIssuance/exchangeIssuance-staged.spec.ts

```diff
@@ -0,0 +1,906 @@
+        const initSushiswapToken1Allownace = await subjectToken1.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+        const initIssuanceModuleToken1Allownace = await subjectToken1.allowance(exchangeIssuance.address, subjectBasicIssuanceModuleAddress);
+
+        const initUniswapToken2Allownace = await subjectToken2.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
+        const initSushiswapToken2Allownace = await subjectToken2.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+        const initIssuanceModuleToken2Allownace = await subjectToken2.allowance(exchangeIssuance.address, subjectBasicIssuanceModuleAddress);
+
+        await subject();
+
+        const finalUniswapToken1Allownace = await subjectToken1.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
+        const finalSushiswapToken1Allownace = await subjectToken1.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+        const finalIssuanceModuleToken1Allownace = await subjectToken1.allowance(exchangeIssuance.address, subjectBasicIssuanceModuleAddress);
+
+        const finalUniswapToken2Allownace = await subjectToken2.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
+        const finalSushiswapToken2Allownace = await subjectToken2.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+        const finalIssuanceModuleToken2Allownace = await subjectToken2.allowance(exchangeIssuance.address, subjectBasicIssuanceModuleAddress);
+
+        expect(finalUniswapToken1Allownace.sub(initUniswapToken1Allownace)).eq(MAX_UINT_96);
+        expect(finalSushiswapToken1Allownace.sub(initSushiswapToken1Allownace)).eq(MAX_UINT_96);
+        expect(finalIssuanceModuleToken1Allownace.sub(initIssuanceModuleToken1Allownace)).eq(MAX_UINT_96);
+
+        expect(finalUniswapToken2Allownace.sub(initUniswapToken2Allownace)).eq(MAX_UINT_96);
+        expect(finalSushiswapToken2Allownace.sub(initSushiswapToken2Allownace)).eq(MAX_UINT_96);
+        expect(finalIssuanceModuleToken2Allownace.sub(initIssuanceModuleToken2Allownace)).eq(MAX_UINT_96);
+      });
+    });
+
+    describe("#issueSetForExactToken", async () => {
+      let subjectCaller: Account;
+      let subjectSetToken: SetToken;
+      let subjectInputToken: StandardTokenMock;
+      let subjectAmountInput: BigNumber;
+      let subjectMinSetReceive: BigNumber;
+
+      beforeEach(async () => {
+        // Deploy any required dependencies
+        subjectCaller = user;
+        subjectSetToken = setToken;
+        subjectInputToken = usdc;
+        subjectAmountInput = BigNumber.from(1000 * 10 ** 6);
+        subjectMinSetReceive = ether(0);
+      });
+
+      async function subject(): Promise<ContractTransaction> {
+        await exchangeIssuance.approveSetToken(subjectSetToken.address);
+        await subjectInputToken.connect(subjectCaller.wallet).approve(exchangeIssuance.address, MAX_UINT_256);
```

Comment:
Move this setup outside of the subject. That way, you can also write a test for when the user does NOT grant the proper approvals.